darts/dataprocessing/transformers/missing_values_filler.py

```diff
@@ -0,0 +1,50 @@
+"""
+Missing Values Filler
+---------------------
+"""
+from typing import Optional, List, Union
+
+from darts.dataprocessing import Validator
+from darts.dataprocessing.transformers import BaseDataTransformer
+from darts.utils.missing_values import auto_fillna, fillna
+
+from darts.timeseries import TimeSeries
+from darts.logging import get_logger, raise_if, raise_if_not
+
+logger = get_logger(__name__)
+
+
+class MissingValuesFiller(BaseDataTransformer[TimeSeries]):
+    def __init__(self,
+                 fill: Union[str, float] = 'auto',
+                 name: str = "MissingValuesFiller",
+                 validators: Optional[List[Validator]] = None):
```

Comment:
I'm open to discussion on the fill parameter type. I've also considered making it optional, and default to calling `auto_fillna()` if not provided / `None`, what do you think is best ?

Also I'm wondering two things about `fillna()` and `auto_fillna()` in `statistics.py`:
- I know they're named after the corresponding pandas methods, but I don't think the name is supper obvious… I'd suggest refactoring both methods into a single one, maybe called `fill_missing_values()` which would then also take a `fill: Union[str, float] = 'auto'` parameter, what do you think ?
- Shouldn't we prefix them with an underscore (make them "private"), to encourage using the transformer instead  ?

Reply:
Cool.

I also think we can remove the `missing_values.py` file and:
- Move `_fill_missing_values()` inside of the `TimeSeries` class. New syntax would be `series._fill_missing_values(fill=…)` instead of `_fill_missing_values(series, fill=…)`

- Rename `na_ratio()` into `missing_values_ratio()` and move it to `statistics.py`

WDYT ?

Reply:
Also questioning this:

> Shouldn't we prefix them with an underscore (make them "private") or even remove them once the transformer is implemented, to encourage using the transformer instead ?

I think that it's more intuitive to be able to simply call `fill_missing_values()` when that is the only transformation the user wants to make, instead of `MissingValuesFiller().transform()`. Of course having the transformer is still very useful because it allows using it in a `Pipeline`.

Thus I think that the best way to go about it is to provide both options.

Reply:
Agree with your EDIT ;)